include/quda.h

```diff
@@ -1039,12 +1046,27 @@ extern "C" {
    *
    * @param _hp_x    Array of solution spinor fields
    * @param _hp_b    Array of source spinor fields
-   * @param param  Contains all metadata regarding
    * @param param  Contains all metadata regarding host and device
    *               storage and solver parameters
    */
+
   void invertMultiSrcQuda(void **_hp_x, void **_hp_b, QudaInvertParam *param);
 
+  /**
+   * @brief Perform the solve like @invertQuda but for multiple rhs by spliting the comm grid into
+   * sub-partitions: each sub-partition invert one rhs.
+   * The QudaInvertParam object specifies how the solve should be performed on each sub-partition.
+   * Unlike @invertQuda and @invertMultiSrcQuda, the interface also takes the host side gauge as
+   * input - gauge field is not required to be loaded beforehand.
+   * @param _hp_x       Array of solution spinor fields
+   * @param _hp_b       Array of source spinor fields
+   * @param param       Contains all metadata regarding host and device storage and solver parameters
+   * @param h_gauge     Base pointer to host gauge field (regardless of dimensionality)
+   * @param gauge_param Contains all metadata regarding host and device storage for gauge field
+   */
+  void invertSplitGridQuda(void **_hp_x, void **_hp_b, QudaInvertParam *param, void *h_gauge,
+                           QudaGaugeParam *gauge_param);
```

Comment:
From a logical point of view, why do we need this extra interface?  Could it be merged with the `invertMultiSrcQuda` interface?

Reply:
Because we need `void *h_gauge` and `QudaGaugeParam *gauge_param`?

Reply:
How about extending `invertMultiSrcQuda` with the additional arguments, and only do the split-grid if these extra pointers are non-zero?  Then we keep a single interface that can be used for all different multi-src solvers, whether fully block, split-grid or hybrid?

Reply:
Now I would really say `invertMultiSrcQuda` is a copy of `invertQuda` but everything acts on multi-rhs. `invertQuda` has been updated by many new features, and these features haven't been added to `invertMultiSrcQuda`. I would propose to unify `invertMultiSrcQuda` and `invertQuda` in a sensible way, add then combining `invertMultiSrcQuda` and `invertSplitGridQuda`. But easier said than done.

Reply:
I wonder if we should just delete `invertMultiSrcQuda` and rename `invertSplitGridQuda` to `invertMultiSrcQuda`, and when we add back in block solvers etc, we would do it there.  Is there any functionality in `invertQuda` that isn't it in `invertSplitGridQuda` yet?  could `invertQuda` just become a thin wrapper around what is currently  `invertSplitGridQuda`?